src/Jupyter/Extensions.cs

```diff
@@ -73,72 +74,88 @@ public static string ToHumanReadableBytes(this long nBytes)
             return $"{nBytes} B";
         }
 
+
         /// <summary>
-        ///     Adds functionality to a given quantum simulator to display
-        ///     diagnostic output with rich Jupyter formatting.
+        ///     An enumerable source of the significant amplitudes of this state
+        ///     vector and their labels, where significance and labels are
+        ///     defined by the values loaded from <paramref name="configurationSource" />.
         /// </summary>
-        /// <param name="simulator">
-        ///     The simulator to be augmented with Jupyter display
-        ///     functionality.
-        /// </param>
-        /// <param name="channel">
-        ///     The Jupyter display channel to be used to display diagnostic
-        ///     output.
-        /// </param>
-        /// <param name="configurationSource">
-        ///      A source of configuration options to be used to set display
-        ///      preferences. Typically, this will be provided by the service
-        ///      provider configured when an execution engine is constructed.
-        /// </param>
-        /// <returns>
-        ///     The value of <paramref name="simulator" />.
-        /// </returns>
-        public static QuantumSimulator WithJupyterDisplay(this QuantumSimulator simulator, IChannel channel, IConfigurationSource configurationSource)
-        {
-            // First, we disable console-based logging so as to not
-            // duplicate messages.
-            simulator.DisableLogToConsole();
-            // Next, we attach the display channel's standard output handling
-            // to the log event.
-            simulator.OnLog += channel.Stdout;
+        public static IEnumerable<(Complex, string)> SignificantAmplitudes(
+            this QuantumSimulator.DisplayableState state, 
```

Comment:
As mentioned in the other PR, I do think it'd be good to make DisplayableState not be an inner type of `QuantumSimulator` so that it's easier to reuse with the sparse simulator.

Reply:
Yes, I remember that from the offline discussion. Looks like I will need to do the stage 3 in parallel in qsharp-runtime repo and iqsharp repo.
I'll leave this comment "Unresolved" until it is resolved.

Reply:
Resolved in a [repeated attempt](https://github.com/microsoft/qsharp-runtime/pull/898).